### ISSUE TITLE
Registration/Tech Debt: Fix playwright accessibility issues in Registration workflow e2e tests

### ISSUE THREAD

**Issue by BCerki** [bcgov/cas-registration#3198]
#### Description of the Tech Debt

In bciers/apps/registration-e2e/tests/register-an-operation.spec.ts, we commented out the `analyzeAccessibility` check where it was failing. This ticket is about fixing the failures and re-enabling the check.

**In the Register an Existing LFO test:**
```
Array [
+   Object {
+     "description": "Ensures every ARIA progressbar node has an accessible name",
+     "help": "ARIA progressbar nodes must have an accessible name",
+     "helpUrl": "https://dequeuniversity.com/rules/axe/4.9/aria-progressbar-name?application=playwright",
+     "id": "aria-progressbar-name",
+     "impact": "serious",
+     "nodes": Array [
+       Object {
+         "all": Array [],
+         "any": Array [
+           Object {
+             "data": null,
+             "id": "aria-label",
+             "impact": "serious",
+             "message": "aria-label attribute does not exist or is empty",
+             "relatedNodes": Array [],
+           },
+           Object {
+             "data": null,
+             "id": "aria-labelledby",
+             "impact": "serious",
+             "message": "aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty",
+             "relatedNodes": Array [],
+           },
+           Object {
+             "data": Object {
+               "messageKey": "noAttr",
+             },
+             "id": "non-empty-title",
+             "impact": "serious",
+             "message": "Element has no title attribute",
+             "relatedNodes": Array [],
+           },
+         ],
+         "failureSummary": "Fix any of the following:
+   aria-label attribute does not exist or is empty
+   aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
+   Element has no title attribute",
+         "html": "<span class=\"MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary mui-1sgvys3-MuiCircularProgress-root\" role=\"progressbar\" style=\"width: 40px; height: 40px;\">",
+         "impact": "serious",
+         "none": Array [],
+         "target": Array [
+           ".MuiCircularProgress-root",
+         ],
+       },
+     ],
+     "tags": Array [
+       "cat.aria",
+       "wcag2a",
+       "wcag111",
+       "EN-301-549",
+       "EN-9.1.1.1",
+     ],
+   },
+   Object {
+     "description": "Ensures elements with an ARIA role that require child roles contain them",
+     "help": "Certain ARIA roles must contain particular children",
+     "helpUrl": "https://dequeuniversity.com/rules/axe/4.9/aria-required-children?application=playwright",
+     "id": "aria-required-children",
+     "impact": "critical",
+     "nodes": Array [
+       Object {
+         "all": Array [],
+         "any": Array [
+           Object {
+             "data": Object {
+               "messageKey": "unallowed",
+               "values": "[role=progressbar]",
+             },
+             "id": "aria-required-children",
+             "impact": "critical",
+             "message": "Element has children which are not allowed: [role=progressbar]",
+             "relatedNodes": Array [
+               Object {
+                 "html": "<span class=\"MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary mui-1sgvys3-MuiCircularProgress-root\" role=\"progressbar\" style=\"width: 40px; height: 40px;\">",
+                 "target": Array [
+                   ".MuiCircularProgress-root",
+                 ],
+               },
+             ],
+           },
+         ],
+         "failureSummary": "Fix any of the following:
+   Element has children which are not allowed: [role=progressbar]",
+         "html": "<div class=\"MuiDataGrid-main mui-204u17-MuiDataGrid-main\" role=\"grid\" aria-colcount=\"4\" aria-rowcount=\"3\" aria-multiselectable=\"false\">",
+         "impact": "critical",
+         "none": Array [],
+         "target": Array [
+           ".MuiDataGrid-main",
+         ],
+       },
+     ],
+     "tags": Array [
+       "cat.aria",
+       "wcag2a",
+       "wcag131",
+       "EN-301-549",
+       "EN-9.1.3.1",
+     ],
+   },
+ ]
```

**In Register a New SFO** (this only happens in CI and playwright flagged as flaky, so it may be testing before the page is fully loaded and in fact there are no violations):
```
Array [
    +   Object {
    +     "description": "Ensures the contrast between foreground and background colors meets WCAG 2 AA minimum contrast ratio thresholds",
    +     "help": "Elements must meet minimum color contrast ratio thresholds",
    +     "helpUrl": "https://dequeuniversity.com/rules/axe/4.9/color-contrast?application=playwright",
    +     "id": "color-contrast",
    +     "impact": "serious",
    +     "nodes": Array [
    +       Object {
    +         "all": Array [],
    +         "any": Array [
    +           Object {
    +             "data": Object {
    +               "bgColor": "#105058",
    +               "contrastRatio": 2.55,
    +               "expectedContrastRatio": "4.5:1",
    +               "fgColor": "#658f94",
    +               "fontSize": "10.5pt (14px)",
    +               "fontWeight": "normal",
    +               "messageKey": null,
    +             },
    +             "id": "color-contrast",
    +             "impact": "serious",
    +             "message": "Element has insufficient color contrast of 2.55 (foreground color: #658f94, background color: #105058, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1",
    +             "relatedNodes": Array [
    +               Object {
    +                 "html": "<div class=\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root mui-1actny7\" role=\"alert\" direction=\"up\" style=\"opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\">",
    +                 "target": Array [
    +                   ".MuiPaper-root",
    +                 ],
    +               },
    +               Object {
    +                 "html": "<div class=\"bg-bc-primary-blue relative left-1/2 transform -translate-x-1/2 w-screen max-w-none\">",
    +                 "target": Array [
    +                   "footer > .bg-bc-primary-blue.w-screen.max-w-none",
    +                 ],
    +               },
    +             ],
    +           },
    +         ],
    +         "failureSummary": "Fix any of the following:
    +   Element has insufficient color contrast of 2.55 (foreground color: #658f94, background color: #105058, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1",
    +         "html": "<div class=\"MuiSnackbarContent-message mui-1w0ym84\">Operation Representative saved successfully</div>",
    +         "impact": "serious",
    +         "none": Array [],
    +         "target": Array [
    +           ".MuiSnackbarContent-message",
    +         ],
    +       },
    +     ],
    +     "tags": Array [
    +       "cat.color",
    +       "wcag2aa",
    +       "wcag143",
    +       "TTv5",
    +       "TT13.c",
    +       "EN-301-549",
    +       "EN-9.1.4.3",
    +       "ACT",
    +     ],
    +   },
    + ]
```

#### Tech Debt Triage

The purpose of our technical debt triage process is to analyze technical debt to determine risk level of the technical debt and the value in tackling that technical debt.

#### Risk Value Scoring:

| Level  | Value                 |
| ------ | --------------------- |
| High   | <div align="center">3 |
| Medium | <div align="center">2 |
| Low    | <div align="center">1 |

| Technical Debt - Risk Types                                                                                                   | Level | Value |
| ----------------------------------------------------------------------------------------------------------------------------- | ----- | ----- |
| Business Area Risk - Risk of business area visibility / damage to user experience                                             | 0     | 0     |
| Developer Fault Risk - How likely will this tech debt cause a future error related to coding on top of it                     | 0     | 0     |
| System Fault Risk - Risk of system errors or application downtime                                                             | 0     | 0     |
| Time Scale Risk - Compound risk effect if left alone. How much more difficult to fix or dangerous will this become over time? | 0     | 0     |
| Time Sink Risk - How much will this tech debt slow the development process down                                               | 0     | 0     |
| <div align="right">**TOTAL SCORE:**                                                                                           | 0     | 0     |

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item

**Comment by NicoleGovvy** [comment]
@BCerki what were the specific issues found here? I'm not sure how to decipher this ticket 🔍 ?

**Comment by fviduya** [comment]
Putting this out here: From the accessibility review we had before, I remember that we are following WCAG 2.2 AA. Maybe we can limit the scope of accessibility here by using tags, here's what I'm thinking off:

- 'wcag2aa' - for the foundational accessibility stuff
- 'wcag22aa' - to include the 2.2 AA ones we have identified during accessibility review
- 'best-practice' - best practices that are not covered by WCAG

Then if the impact is not 'serious' or 'critical', we can bypass and log the issue found.

**Comment by gdalcengio** [comment]
Registration seems to be fine, it's administration that's having accessibility issues
